### PR TITLE
Prepare for setting PERMANENTLY_DOWN

### PR DESCRIPTION
**Pull request by hakonhall**

### CODE REVIEW

File: orchestrator/src/main/java/com/yahoo/vespa/orchestrator/status/ZookeeperStatusService.java [hunk 1]
```diff
@@ -99,12 +99,13 @@ public boolean setHostStatus(ApplicationInstanceReference application, HostName
 
     /**
      * Cache is checked for freshness when this mapping is created, and may be invalidated again later
-     * by other users of the cache. Since this function is backed by the cache, any such invalidations
+     * by other users of the cache. Since this function is backed by the cache, any such invalidation
      * will be reflected in the returned mapping; all users of the cache collaborate in repopulating it.
      */
     @Override
-    public Function<ApplicationInstanceReference, Set<HostName>> getSuspendedHostsByApplication() {
-        return application -> hostInfosCache.getHostInfos(application).suspendedHostsnames();
```
Comment (hakonhall) [inline orchestrator/src/main/java/com/yahoo/vespa/orchestrator/status/ZookeeperStatusService.java]: This line is fairly new. The older code read the cache counter only once, and cleared the cache only once (or didn't clear it at all). While the current code will do cache invalidation on every invocation of the returned Function. The old behavior is restored with this PR.

File: orchestrator/src/main/java/com/yahoo/vespa/orchestrator/status/ZookeeperStatusService.java [hunk 2]
```diff
@@ -99,12 +99,13 @@ public boolean setHostStatus(ApplicationInstanceReference application, HostName
...
+    public Function<ApplicationInstanceReference, HostInfos> getHostInfosByApplicationResolver() {
+        hostInfosCache.refreshCache();
+        return hostInfosCache::getCachedHostInfos;
```
Comment (freva) [inline orchestrator/src/main/java/com/yahoo/vespa/orchestrator/status/ZookeeperStatusService.java]: Why not just 
```suggestion
        return hostInfosCache::getHostInfos;
```
?

**Review by freva — APPROVED** [review]

File: orchestrator/src/main/java/com/yahoo/vespa/orchestrator/status/ZookeeperStatusService.java (hunk 2 above)
Comment (hakonhall) [reply orchestrator/src/main/java/com/yahoo/vespa/orchestrator/status/ZookeeperStatusService.java]: I tried to explain that to the left?

Comment (freva) [reply orchestrator/src/main/java/com/yahoo/vespa/orchestrator/status/ZookeeperStatusService.java]: :man_facepalming:

**Pull request merged**